src/ppx/xr_component.cpp

```diff
@@ -253,6 +300,74 @@ void XrComponent::Destroy()
     }
 }
 
+void XrComponent::BeginPassthrough()
+{
+    if (!mPassthroughSupported) {
+        return;
+    }
+
+#if defined(PPX_XR_QUEST)
+    PFN_xrPassthroughStartFB pfnXrPassthroughStartFB = nullptr;
+    CHECK_XR_CALL(xrGetInstanceProcAddr(mInstance, "xrPassthroughStartFB", (PFN_xrVoidFunction*)(&pfnXrPassthroughStartFB)));
+    PPX_ASSERT_MSG(pfnXrPassthroughStartFB != nullptr, "Cannot get xrPassthroughStartFB function pointer!");
+    CHECK_XR_CALL(pfnXrPassthroughStartFB(mPassthrough));
+
+    if (!mPassthroughLayer) {
+        XrPassthroughFlagsFB           layerFlags                    = {};
+        XrPassthroughLayerCreateInfoFB passthroughLayerInfo          = {XR_TYPE_PASSTHROUGH_LAYER_CREATE_INFO_FB};
+        passthroughLayerInfo.next                                    = nullptr;
+        passthroughLayerInfo.passthrough                             = mPassthrough;
+        passthroughLayerInfo.flags                                   = layerFlags;
+        passthroughLayerInfo.purpose                                 = XR_PASSTHROUGH_LAYER_PURPOSE_RECONSTRUCTION_FB;
+        PFN_xrCreatePassthroughLayerFB pfnXrCreatePassthroughLayerFB = nullptr;
+        CHECK_XR_CALL(xrGetInstanceProcAddr(mInstance, "xrCreatePassthroughLayerFB", (PFN_xrVoidFunction*)(&pfnXrCreatePassthroughLayerFB)));
+        PPX_ASSERT_MSG(pfnXrCreatePassthroughLayerFB != nullptr, "Cannot get xrCreatePassthroughLayerFB function pointer!");
+        CHECK_XR_CALL(pfnXrCreatePassthroughLayerFB(mSession, &passthroughLayerInfo, &mPassthroughLayer));
+        PPX_ASSERT_MSG(mPassthroughLayer != XR_NULL_HANDLE, "XrPassthroughLayerFB creation failed!");
+    }
+
+    PFN_xrPassthroughLayerResumeFB pfnXrPassthroughLayerResumeFB = nullptr;
+    CHECK_XR_CALL(xrGetInstanceProcAddr(mInstance, "xrPassthroughLayerResumeFB", (PFN_xrVoidFunction*)(&pfnXrPassthroughLayerResumeFB)));
+    PPX_ASSERT_MSG(pfnXrPassthroughLayerResumeFB != nullptr, "Cannot get xrPassthroughLayerResumeFB function pointer!");
+    CHECK_XR_CALL(pfnXrPassthroughLayerResumeFB(mPassthroughLayer));
+#endif // defined(PPX_XR_QUEST)
+
+    mPassthroughEnabled = true;
```

Comment:
shouldn't this be set to true only in the PPX_XR_QUEST case? (shouldn't reach that, but to make code coherent) (same in End function)